### DISCUSSION TITLE
Is there any way to access event.context.params in a server middleware?

### DISCUSSION THREAD

**Discussion by ad-on-is** [nuxt/nuxt#22152] [Questions]
Basically the title!

I have the following structure, and the middleware does not seem to be aware of the event.context.params

/server/api/clients/[id].ts -> `event.context` has a **params** property populated
/server/middleware/mymiddleware.ts -> `event.context` has NO **params** property

Am I doing something wrong?

**Comment by arashsheyda** [comment]
I think the `event.context.params` property is only available in the route handler, not in the middleware. This is because the middleware is executed before the route handler, so it doesnt't have access to the route parameters.

but you can access the url with `getRequestURL(event)` (or use `getRequestPath(event)`), and then extract the param from url maybe?

**Reply by ad-on-is** [reply]
Not an option, since what I need to do relies on the property names.

Example: I encode the ID property of a database result, so {id:1} becomes {id:'Ck9ag5'} etc...

users/[id].ts 
clients/[id].ts
...etc

I need to encode all the IDs back, before they hit the route, so I can use them for further database queries.

The only way I found so far is to use event.context.params = encodeIDs(event.context.params) in each route.
